detectors/node/opentelemetry-resource-detector-container/src/detectors/ContainerDetector.ts

```diff
@@ -57,9 +57,29 @@ export class ContainerDetector implements Detector {
       this.UTF8_UNICODE
     );
     const splitData = rawData.trim().split('\n');
-    for (const str of splitData) {
-      if (str.length >= this.CONTAINER_ID_LENGTH) {
-        return str.substring(str.length - this.CONTAINER_ID_LENGTH);
+    for (const line of splitData) {
+      const lastSlashIdx = line.lastIndexOf('/');
+      if (lastSlashIdx === -1){
+        continue;
+      }
```

Comment:
looks like there was just some missing whitespace ` `
```suggestion
      if (lastSlashIdx === -1) {
        continue;
      }
```